app/javascript/packs/hideBookmarkButtons.js

```diff
@@ -1,12 +1,19 @@
 import { getUserDataAndCsrfToken } from '@utilities/getUserDataAndCsrfToken';
 
-getUserDataAndCsrfToken().then(({ currentUser }) => {
-  const currentUserId = currentUser && currentUser.id;
+getUserDataAndCsrfToken().then(
+  ({ currentUser }) => {
+    const currentUserId = currentUser && currentUser.id;
 
-  document.querySelectorAll('.bookmark-button').forEach((button) => {
-    const { articleAuthorId } = button.dataset;
-    if (currentUserId && articleAuthorId == currentUserId) {
-      button.classList.add('hidden');
+    document.querySelectorAll('.bookmark-button').forEach((button) => {
+      const { articleAuthorId } = button.dataset;
+      if (currentUserId && articleAuthorId == currentUserId) {
+        button.classList.add('hidden');
+      }
+    });
+  },
+  (error) => {
+    if (!/find user data/.test(error)) {
```

Comment:
This feels like something that should live closer to the source? E.g. an option passed to `getUserDataAndCsrfToken` that will suppress the error where it's thrown instead of needing to pattern-match against it like this.

Reply:
I tried to move it closer to the source (and satisfy Codecov) in [1c6b9b4](https://github.com/forem/forem/pull/19360/commits/1c6b9b400d0dffeb4498bb5d647a01bc606121e7) — want to re-evaluate?

Reply:
Yep, that seems a lot clearer in its intent